### DISCUSSION TITLE
Run multi Uvicorn API on same server

### DISCUSSION THREAD

**Discussion by allyvan1314** [Kludex/uvicorn#1393] [Q&A] [answered]
Hi, I've search for result on GG but none work so I create a discussion, please help.
I have 5 difference APIs on 5 py files. I've seen that I can run only one API in 1 Console to kernel (in Spyder or Terminal). Is there any way that I can run all 5 APIs at same time?
If I run each line separate it all work fine but if I uncomment and run, the bellow code won't work 

```python
from uvicorn import run
if __name__ == "__main__":
    run("API_B0_MOB123:app", host="127.0.0.1", port=5000, reload=True)
    run("API_B0_GTE4:app", host="127.0.0.1", port=7000, reload=True)
    run("API_B0_GTE6:app", host="127.0.0.1", port=8000, reload=True)
    run("API_B1:app", host="127.0.0.1", port=9000, reload=True)
    run("API_B2:app", host="127.0.0.1", port=3000, reload=True)
```

**Comment by Kludex** [comment]
https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor

**Reply by allyvan1314** [reply]
Sorry for any inconvenient because I haven't use this function before. I read example and see that executor use map() function with function and data. In my case, which variable should I pass in map() function and data? Do I have to declare an params var?

```python
def main():
    with concurrent.futures.ProcessPoolExecutor() as executor:
        for number, prime in zip(PRIMES, executor.map(is_prime, PRIMES)):
            print('%d is prime: %s' % (number, prime))
```

**Comment by euri10** [comment, marked as answer]
Not sure what your goal is but you could supervisord to manage your processes : http://supervisord.org/ 
Containers would be a good solution too
